hybrid-array/src/lib.rs

```diff
@@ -13,7 +13,6 @@
     clippy::cast_sign_loss,
     clippy::checked_conversions,
     clippy::implicit_saturating_sub,
-    clippy::arithmetic_side_effects,
```

Comment:
I'd prefer to keep this on at a crate-level. Perhaps you can disable it at the individual function level or operation level with a `SAFETY`-style comment about why overflow/underflow won't occur?